Prog/Demographic trends.sas

```diff
@@ -0,0 +1,1458 @@
+  if ucounty  in ("51013") then Jurisdiction =6;
+  if ucounty  in ("51059", "51600", "51610") then Jurisdiction =7;
+  if ucounty  in ("51107") then Jurisdiction =8;
+  if ucounty  in ("51153", "51683", "51685") then Jurisdiction =9; 
+  if ucounty  in ("51510") then Jurisdiction =10; 
+
+run;
+
+proc sort data=households;
+by ucounty;
+run;
+/* calculate vacancy rate in 2017 and estimate hh counts based on housing units*/
+data COGSarea_2017 (where=(upuma in ("1100101", "1100102", "1100103", "1100104", "1100105", "2401600", "2400301", "2400302","2401001", "2401002", "2401003", "2401004", "2401005", "2401006", "2401007", "2401101", "2401102", "2401103", "2401104", "2401105", "2401106", "2401107", "5101301", "5101302", "5159301", "5159302", "5159303", "5159304", "5159305", "5159306", "5159307", "5159308", "5159309", "5110701", "5110702" , "5110703", "5151244", "5151245", "5151246", "5151255"
+                    ) and pernum =1 and gq in (1,2)));
+set Ipums.Acs_2017_dc Ipums.Acs_2017_md Ipums.Acs_2017_va;
+keep upuma Jurisdiction hhwt totalunits pernum gq;
+	%assign_jurisdiction; 
+totalunits=1;
+	run;
+
+proc sort data=COGSarea_2017;
+by upuma;
+run;
+proc summary data=COGSarea_2017;
+by upuma;
+var totalunits;
+weight hhwt;
+output out = COGSarea_2017_sum sum=;
+run;
+
+proc sort data=COGSarea_2017_sum;
+by upuma;
+run;
+
+data COGSvacant_2017 (where=(upuma in ("1100101", "1100102", "1100103", "1100104", "1100105", "2401600", "2400301", "2400302","2401001", "2401002", "2401003", "2401004", "2401005", "2401006", "2401007", "2401101", "2401102", "2401103", "2401104", "2401105", "2401106", "2401107", "5101301", "5101302", "5159301", "5159302", "5159303", "5159304", "5159305", "5159306", "5159307", "5159308", "5159309", "5110701", "5110702" , "5110703", "5151244", "5151245", "5151246", "5151255" )));
+set Ipums.Acs_2017_vacant_dc Ipums.Acs_2017_vacant_md Ipums.Acs_2017_vacant_va;
+
+	%assign_jurisdiction; 
+if vacancy in (1,2,3) then vacunit = 1; else vacunit = 0;
+run;
+
+data COGvacant2017;
+set COGSvacant_2017(keep=upuma hhwt VACANCY Jurisdiction vacunit where=(vacancy in (1,2,3)));
+run;
+
+proc sort data= COGvacant2017;
+by upuma;
+run;
+
+proc summary data=COGvacant2017;
+by upuma;
+var vacunit;
+weight hhwt;
+output out = COGvacant2017_sum sum=;
+run;
+
+proc sort data=COGvacant2017_sum;
+by upuma;
+run;
+
+data vacantunits;
+merge COGvacant2017_sum COGSarea_2017_sum;
+by upuma;
+	%assign_jurisdiction; 
+run;
+
+proc summary data= vacantunits;
+	class Jurisdiction;
+	var totalunits vacunit;
+	output out = vacancyrate  sum=;
+	format Jurisdiction Jurisdiction. ;
+	run;
+
+data vacancyrate2;  
+set vacancyrate;
+keep Jurisdiction vacrate;
+vacrate= vacunit/totalunits;
+run;
+
+proc sort data=COGSarea_2017 ;
+by Jurisdiction;
+run;
+
+data hh_housing;
+merge COGSarea_2017 vacancyrate2;
+by Jurisdiction;
+run;
+
+data hhestimates17 (drop= _FREQ_);
+set hh_housing;
+hhestimate17= totalunits*(1-vacrate);
```

Comment:
I think you said that the 2017 population and housing unit counts match the Census estimates, so it doesn't seem like we need this part? And you don't seem to use HHESTIMATE17 anywhere. I would just delete all of this if it is unnecessary.

Reply:
I used it right in the next step? This dataset uses vacancy rate to calculate the estimated household count that is not in NCDB.